coalib/misc/Annotations.py

```diff
@@ -0,0 +1,41 @@
+
+
+def typechain(*args):
+    """
+    Returns function which applies the first transformation it can from args
+    and returns transformed value, raises ValueError when cannot apply none.
+
+    :raises TypeError:  Raises when either no functions are specified for
+                        checking or one or more object isn't callable and not
+                        None.
+
+    >>> function = typechain(int, ord)
+    >>> function("10")
+    10
+    >>> function("a")
+    97
+    >>> function("str")
+    Traceback (most recent call last):
+        ...
+    ValueError: Couldn't convert to any specified type value 'str'.
+
+    """
+    if len(args) == 0:
+        raise TypeError("No arguments were provided.")
+    if not all(arg is None or getattr(arg, '__call__', False) for arg in args):
+        raise TypeError("One of the provided argument is not callable.")
+
+    def annotation(value):
+        for arg in args:
+            if value is arg:
+                return value
+            if isinstance(arg, type) and isinstance(value, arg):
+                return value
+            try:
+                return arg(value)
+            except (ValueError, TypeError):
+                pass
+        raise ValueError(
+            "Couldn't convert to any specified type value "
+            "'{value}'.".format(value=value))
```

Comment:
repr may contain that info when it's suitable, using repr is what we do in such cases and each class should decide individually how it wants to be represented best